mkdocs_git_authors_plugin/repo.py

```diff
@@ -0,0 +1,774 @@
+from pathlib import Path
+import logging
+import os
+import re
+import subprocess
+
+from datetime import datetime, timedelta, timezone
+
+class GitCommandError(Exception):
+    """
+    Exception thrown by a GitCommand.
+    """
+    pass
+
+
+class GitCommand(object):
+    """
+    Wrapper around a Git command.
+
+    Instantiate with a command name and an optional args list.
+    These can later be modified with set_command() and set_args().
+
+    Execute the command with run()
+
+    If successful the results can be read as string lists with
+    - stdout()
+    - stderr()
+    In case of an error a verbose GitCommandError is raised.
+    """
```

Comment:
I just re-read the [Zen of Python](https://www.python.org/dev/peps/pep-0020/)

> There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.

I'm actually Dutch, but this point is not super obvious to me 😞😋. It's probably a matter of preference / coding style. The 'zen' does not mention the large file over one-file-per object.

For me it would be more readable to have a `git/` folder with smaller files for the main classes. But I won't press the issue, I'll leave it to you to decide.

Reply:
OK, maybe my recollection has misled me. Probably I've read some *comments* on the topic.

I agree that factoring out multiple modules may make the code cleaner, and maybe it will even help with one minor issue I had: I couldn't *always* provide type hinting because of circular dependencies between classes.

I agree to doing that, but only after a decision whether the Git classes will actually be kept.

#23